src/accessors.d

```diff
@@ -84,39 +86,53 @@ template GenerateReader(string name, alias field)
     static enum helper()
     {
         import std.string : format;
-        import std.traits : isArray, isSomeString;
 
         enum visibility = getVisibility!(field, Read);
         enum accessorName = accessor(name);
+        enum needToDup = needToDup!field;
 
-        static if (isArray!(typeof(field)) && !isSomeString!(typeof(field)))
+        static if (needToDup)
+        {
+            enum uint postblitAttrs = inferAttributes!(typeof(field[0]), "__postblit") &
+                ~FunctionAttribute.nogc;
+        }
+        else
         {
-            return format("%s final @property auto %s() {"
+            enum uint postblitAttrs = inferAttributes!(typeof(field), "__postblit");
+        }
+
+        string attributes = generateAttributeString!postblitAttrs;
+
+        static if (needToDup)
+        {
+            return format("%s final @property auto %s() %s{"
```

Comment:
always (also in `unittest`) keep the body on one line:

    ~ "{ return [] ~ this.%s; }